auth/README.md

```diff
@@ -203,38 +203,46 @@ startActivityForResult(
 #### Handling the sign-in response
 
 #####Response codes
-The authentication flow only provides three response codes:
-`Activity.RESULT_OK` if a user is signed in, `Activity.RESULT_CANCELLED` if
-sign in failed, and `ResultCodes.RESULT_NO_NETWORK` if sign in failed due to a lack of network connectivity.
-No further information on failure is provided as it is not
-typically useful; the only recourse for most apps if sign in fails is to ask
-the user to sign in again later, or proceed with anonymous sign-in if
-supported.
+The authentication flow provides several response codes of which the most common are as follows:
+`ResultCodes.OK` if a user is signed in, `ResultCodes.CANCELLED` if the user manually canceled the sign in,
+`ResultCodes.NO_NETWORK` if sign in failed due to a lack of network connectivity,
+and `ResultCodes.UNKNOWN_ERROR` for all other errors.
+Typically, the only recourse for most apps if sign in fails is to ask
+the user to sign in again later, or proceed with anonymous sign-in if supported.
 
 ```java
 protected void onActivityResult(int requestCode, int resultCode, Intent data) {
     super.onActivityResult(requestCode, resultCode, data);
-    if (resultCode == RESULT_OK) {
-        // user is signed in!
-        startActivity(new Intent(this, WelcomeBackActivity.class));
-        finish();
-        return;
-    }
+    if (requestCode == RC_SIGN_IN) {
+        if (data != null) {
+            IdpResponse response = IdpResponse.fromResultIntent(data);
```

Comment:
That's cleaner, but it doesn't work. Since we are now returning `ResultCodes.CANCELED` with an added error code, it's a little different. So how about [this](https://github.com/SUPERCILEX/FirebaseUI-Android/blob/d56058b85b02648ed2c4479c699ffe9f6259b6c7/app/src/main/java/com/firebase/uidemo/auth/AuthUiActivity.java#L202)?